sdk/keyvault/azure-security-keyvault/azure/security/keyvault/secrets/README.md

```diff
@@ -0,0 +1,171 @@
+
+Use the secret client library to create and manage secrets.
+
+[Source code](https://github.com/Azure/azure-sdk-for-python/tree/master/sdk/keyvault/azure-security-keyvault/azure/security/keyvault/secrets) | [Package (Pypi)](TODO) | [API reference documentation](TODO) | [Product documentation](TODO)
+## Getting started
+### Install the package
+Install the Azure Key Vault client library for Python with pip:
+
+
+`$ pip install azure-security-keyvault
+`
+
+### Prerequisites
+* An [Azure subscription](https://azure.microsoft.com/free/).
+* Python 2.7, 3.4 or later.
+* An existing Key Vault. You can create this by following the instructions on [this article](https://docs.microsoft.com/en-gb/azure/key-vault/quick-create-portal).
+
+### Authenticate the client
+In order to interact with the Key Vault service for managing secrets, you'll need to create an instance of the SecretClient class. For this you will need a vault url, client secret credentials (client id, client secret, tenant id) and a [resource url](https://vault.azure.net).
+
+### Get credentials
+You can find credential information in [Azure Portal](https://portal.azure.com/).
+
+### Create Secret client
+The following code snippet demonstrates a way you can instantiate the KeyVaultClient object:
+```python
+    from azure.identity import AsyncClientSecretCredential
+    from azure.security.keyvault import SecretClient
+
+    credential = AsyncClientSecretCredential(client_id=client_id, secret=client_secret, tenant_id=tenant_id)
+
+    # Create a new secret client using a client secret credential
+    secret_client = SecretClient(vault_url=vault_url, credential=credential)
+```
+## Key concepts
+### Secret
+  A secret is the fundamental resource within Azure KeyVault. From a developer's perspective, Key Vault APIs accept and return secret values as strings. In addition to the secret data, the following attributes may be specified:
+* expires: Identifies the expiration time on or after which the secret data should not be retrieved.
+* nbf: Identifies the time after which the secret will be active.
+* enabled: Specifies whether the secret data can be retrieved.
+* created: Indicates when this version of the secret was created.(read-only)
+* updated: Indicates when this version of the secret was updated.(read-only)
+
+### Secret Client:
+The Secret client performs the interactions with the Azure Key Vault service for getting, setting, updating,deleting, and listing secrets and its versions. An asynchronous and synchronous, SecretClient, client exists in the SDK allowing for selection of a client based on an application's use case.
+
+## Examples
```

Comment:
We should add an example that returns a list of things as that involves different return types. For example, `get_secrets` and that one is clearly different between sync and async